analyzers/src/SonarAnalyzer.Common/Helpers/SyntaxNodeExtensions.cs

```diff
@@ -33,5 +33,10 @@ public static bool ToStringContainsEitherOr(this SyntaxNode node, string a, stri
             var toString = node.ToString();
             return toString.Contains(a) || toString.Contains(b);
         }
+
+        internal static TSyntaxKind Kind<TSyntaxKind>(this SyntaxNode node) where TSyntaxKind : struct, Enum // internal to not be confused with e.g. CSharp.SyntaxNode.Kind()
```

Comment:
There is an FP code smell reported here. It is already tracked by https://github.com/SonarSource/sonar-dotnet/issues/4548 and should be fixed along with https://github.com/SonarSource/sonar-dotnet/issues/6529